### ISSUE TITLE
Resolve international SMS situation

### ISSUE THREAD

**Issue by secretrobotron** [mozilla/webmaker.org#1393]
From #1288, before MWC, we acquired a Spanish number to send installation SMSs for Webmaker app.

Assuming we're upgrading Webmaker app and that it will still need an install page, we should come up with a longer-term solution.

Short code? Long code? I'm out of my region of expertise.

cc @simonwex @thisandagain

**Comment by jbuck** [comment]
Short codes require 3 months to approve and obtain. We should just support multiple "From" SMS phone numbers

**Comment by simonwex** [comment]
This depends on a number of factors and is not consistent around the globe. 

&lt;braindump&gt;

### tl;dr: 
**Short codes**: Expensive, playing by the rules, memorable, some features you can't get on long-codes
**Long codes**: Cheap, not legit -- can be shut off, can also use for voice termination, can't assume costs for receiver, probably where we want to start

When sending from non-phone accounts the two types of numbers are referred to as long codes and short codes. 

Long codes look just like normal numbers, and in fact they are. You can also terminate voice calls on these numbers. Technically in North America and across some other markets, sending automated messages from a long code is not allowed.

Short codes can be purchased at great cost and custom short codes (vanity codes) at even greater cost. The body that governs this in North-America is the MMA (Mobile Marketing Association). They coordinate with the various operators to provision short codes. The fact that you can have the same short code across multiple carriers is merely a coordination effort. This is tracked at the source in an excel spreadsheet, for real. (rly?)>:turtle: Yes turtle, no joke. The low-tech of this industry is why you can, for the most, part get away with a lot provided you don't generate customer complaints. Short codes must implement a bunch of policies including standard messaging and stop/opt-out functionality. This is all checked as part of the approval process, which can take some time. I have some contacts that could likely help accelerate this if need-be.

|                       | Short Codes                                                                                                                                                       | Long Codes                                                                                                                                        |
|-----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|
| Volume                | Sky is the limit, when provisioning a "bind" (read: carrier connection) the max number of messages per second is negotiated                                       | There's a limit in both monthly volume and burst rates, using someone like Twillio likely throttles this for us                     |
| Branding              |  :8ball:  Easier to remember<br> :8ball:  We "own" it<br> :8ball:  Can be coordinated across borders, but it's tough and per-carrier<br> :8ball: Would be interesting to collaborate with MoCo on this | :8ball: Numbers are typical length of MINs<br> :8ball:  Can be discontinued due to complaints                                                                    |
| Voice                 | No voice functionality                                                                                                                                            | Can use to originate or terminate voice calls                                                                                                     |
| Cost for us           | Yikes                                                                                                                                                             | Service provider cost (low)                                                                                                                       |
| End-user cost         | We can, in some countries assume delivery costs                                                                                                                   | End-users pay to receive messages as part of their regular account. In some countries this can be quite expensive to receive international texts. |
| International sending | You can only send to customers on a carrier the short code has been approved on.                                                                                  |                                                                                                                                                   |
